doc/metadata-server/README.md

```diff
@@ -0,0 +1,327 @@
+## Abstract
+
+This document summarizes requirements and proposes a draft of a protocol for metadata information exchange for the on-chain transactions (`Tx`) which executes Marlowe contracts. The off-chain metadata storage mechanism does not rely on a centralized registry but provides schema which can be implemented by independent metadata providers.
+
+## Motivation
+
+Marlowe is a high-level language with a set of properties which are crucial for writing reliable financial contracts. A possibility to audit the code of the contract which was already initiated on-chain seems to be critical to any participant "invited" to it. Marlowe and its current implementation on Cardano chain gives us access to the source code of the contract, ability to do static analysis over it and "step by step" simulation so it provides the fundamental layer of trust for the language users.
+
+To build a contract exchange platform (Marlowe Marketplace) and execution environment (Marlowe Run) we want to be able to also associate and provide contract template information suitable for human consumption (like title, description, statistics and certification etc.). We want to verify that provided metadata information about the transaction template is consistent with the Marlowe contract embedded in the transaction on-chain. Additionally we should allow the contract initiator to attach a specific set of data (like documents, agreements) to the particular contract instance. We want to be able to provide a validity proof for these data and present it to all other participants as well if necessary.
+
+We want to implement this in an efficient manner avoiding huge payloads on the blockchain and by use of existing or emerging Cardano standards. Additionally we need to keep the protocol decentralized so anybody can take the role of a metadata provider.
+
+### Scope limits
+
+We can identify additional sources of metadata information which can be associated with contract transactions running on the blockchain. There are also parts of the proposed metadata format which we should extend in the future. These parts/problems we want to tackle in a separate proposals:
+
+- Role token metadata (icon, currency description etc.) handling. They are presented in Cardano wallets and blockchain explorers.
+
+- Localization (`l10n`) layer for the metadata format.
+
+- Separation of roles / choices identifiers from the presentation layer. These identifiers should have representation in the metadata part as well.
+
+- Encryption layer. We assume now that exchanged data are public or that the access to the information is restricted by a separate layer of the metadata server. In the future we want to probably propose some encryption schema as a part of the protocol.
+
+- Proposal for "identity layer" using Atala PRISM. With identities we can associate and exchange some metadata information like nicknames, credibility etc.
+
+## Verifying Marlowe Contract Metadata on the chain
+
+### Terminology
```

Comment:
I would also leave terminology to the end. Best to explain it as you need it. Could add a glossary at the end for completion

Reply:
Also there are three types of metadata and I cannot tell the difference. I think it could be more useful to the reader to have a discussion of how they are different instead

Reply:
I would like to leave the terminology here and explain the difference using it below.